modules/apps/transfer/keeper/keeper.go

```diff
@@ -169,8 +169,14 @@ func (k Keeper) SetTotalEscrowForDenom(ctx sdk.Context, coin sdk.Coin) {
 	}
 
 	store := ctx.KVStore(k.storeKey)
-	bz := k.cdc.MustMarshal(&sdk.IntProto{Int: coin.Amount})
-	store.Set(types.TotalEscrowForDenomKey(coin.Denom), bz)
+	key := types.TotalEscrowForDenomKey(coin.Denom)
+
+	if coin.Amount.IsZero() {
+		store.Delete(key)
+	} else {
+		bz := k.cdc.MustMarshal(&sdk.IntProto{Int: coin.Amount})
+		store.Set(key, bz)
+	}
```

Comment:
nit: we could return early and omit the `else`, maybe its more "idiomatic" go.

I don't feel too strongly about it here tho

```suggestion
	if coin.Amount.IsZero() {
		store.Delete(key)
		return
	}
	
	bz := k.cdc.MustMarshal(&sdk.IntProto{Int: coin.Amount})
	store.Set(key, bz)
```